sdk/core/Azure.Core/src/ResourceIdentifier.cs

```diff
@@ -13,167 +13,243 @@ namespace Azure.Core
     /// </summary>
     public sealed class ResourceIdentifier : IEquatable<ResourceIdentifier>, IComparable<ResourceIdentifier>
     {
+        private enum SpecialType
+        {
+            Subscription,
+            ResourceGroup,
+            Location,
+            Provider,
+            None
```

Comment:
Consider making None the first element. That way it will map to `0` and allow additional values to be added to this if needed.